### PR TITLE
Test 201 Team administration

### PR DESCRIPTION
**Pull request by stoyanov-ivelin**
Frontend - Added Teams component where the user can see a list of all teams and the user in them, create a new team, delete an empty team and see all users without a team.

Backend - Added endpoint to get all users with their position and team prop as object, endpoint to create and delete a team.

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -102,6 +103,12 @@ class App extends Component<AppProps, AppState> {
                 isAuthenticated={this.isUserAuthenticated(UserRolesEnum.admin)}
                 component={UsersList}
               />
+              <PrivateRoute
+                path="/admin/teams"
+                exact
+                isAuthenticated={this.props.user.isAuthenticated}
```
Comment (kristiyanplpeev) [inline client/src/App.tsx:109]: Use the method isUserAuthenticated here

File: server/src/model/user.entity.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export class Userdb extends BaseEntity {
       lastName: this.lastName,
       picture: this.picture,
       role: this.role,
-      team: this.team,
+      team: this.team?.toTeams(),
       position: this.position?.toPositions(),
```
Comment (kristiyanplpeev) [inline server/src/model/user.entity.ts:48]: Please change this lines to this.team ? this.team.toTeams() : null, same logic to position

**Review by kristiyanplpeev — CHANGES_REQUESTED** [review]

File: client/src/App.tsx (hunk 1 above)
Comment (stoyanov-ivelin) [reply client/src/App.tsx:109]: done

File: server/src/model/user.entity.ts (hunk 2 above)
Comment (stoyanov-ivelin) [reply server/src/model/user.entity.ts:48]: done

**Review by kristiyanplpeev — APPROVED** [review]

**Review by MilenIvanovv — CHANGES_REQUESTED** [review]
The code looks great! Left some comments where I think it can be improved. 

I have a question:
What happens when you delete a team and the team has members?

File: client/src/components/Teams/Teams.tsx [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+    };
+  }
+
+  async componentDidMount(): Promise<void> {
+    this.setState({
+      loading: true,
+    });
+
+    await this.loadTeams();
+    await this.loadUsers();
```
[the remark this answers is not in the document]
Comment (stoyanov-ivelin) [reply client/src/components/Teams/Teams.tsx:73]: done

File: client/src/components/Teams/Teams.tsx [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+          <Button variant="outlined" color="error" className="modal-buttons" onClick={this.handleDeleteModalClose}>
+            Cancel
+          </Button>
+        </Box>
+      </Modal>
+    );
+  }
+
+  getSortedUsers(users: Array<IUserWithTeamAndPosition>): Array<IUserWithTeamAndPosition> {
+    return [...users].sort((a, b) => {
```
[the remark this answers is not in the document]
Comment (stoyanov-ivelin) [reply client/src/components/Teams/Teams.tsx:258]: done

File: client/src/components/Teams/Teams.tsx [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+    }
+
+    return false;
+  };
+
+  handleDelete = async (teamId: string): Promise<void> => {
+    try {
+      await this.userService.deleteTeam(teamId);
+
+      const remainingTeams = this.state.teams.filter((t) => t.id !== teamId);
```
[the remark this answers is not in the document]
Comment (stoyanov-ivelin) [reply client/src/components/Teams/Teams.tsx:299]: done

**Review by MilenIvanovv — APPROVED** [review]

**Pull request merged**